### DISCUSSION TITLE
Persistent structures not detected.

### DISCUSSION THREAD

**Discussion by mattiolabs** [fbarresi/TwinCatAdsTool#16] [Show and tell]
Good morning, I created a project with a structure variable, formed in turn by 2 structures: 1 persistent and 1 not.
![2021-02-12_10h45_46](https://user-images.githubusercontent.com/15252335/107754205-70474780-6d21-11eb-81bd-10dea4a25638.png)

However, the ADS tool does not detect it.
![2021-02-12_10h47_54](https://user-images.githubusercontent.com/15252335/107754219-76d5bf00-6d21-11eb-93c3-5b5b14451ac1.png)

However, the structure is maintained on restart, then managed by TC.

**Comment by fbarresi** [comment]
Ciao Andrea!

`Grazie per aver iniziato una nuova discussione!`

Thank you for starting this new discussion. 
You're right. The software actually only looks for **global** persistent variables and doesn't see such partial persistent objects. 
That's already was reported into #14 and I'm already thinking about a solution.

So stay tuned!

Best regards

`Tanti saluti!`

FB

**Comment by fbarresi** [comment]
Hi Andrea!
I hope you are well and healthy.
I made a pre-release of the software where non global fields are retrieved. 
A pre-release is available [here](https://github.com/fbarresi/TwinCatAdsTool/releases/tag/1.2.10).
I think it would be fine if you would be my beta-tester and give this pre-release a try and share your impressions or open more issues.
Best regards, 
FB
